src/plugins/data/server/search/expressions/esaggs.ts

```diff
@@ -0,0 +1,107 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { get } from 'lodash';
+import { i18n } from '@kbn/i18n';
+import { KibanaRequest, StartServicesAccessor } from 'src/core/server';
+import { Adapters } from 'src/plugins/inspector/common';
+import {
+  EsaggsExpressionFunctionDefinition,
+  EsaggsStartDependencies,
+  getEsaggsMeta,
+  handleEsaggsRequest,
+} from '../../../common/search/expressions';
+import { DataPluginStartDependencies, DataPluginStart } from '../../plugin';
+
+/** @internal */
+function createEsaggs({
+  getStartDependencies,
+}: {
+  getStartDependencies: (req: KibanaRequest) => Promise<EsaggsStartDependencies>;
+}) {
+  return (): EsaggsExpressionFunctionDefinition => ({
+    ...getEsaggsMeta(),
+    async fn(input, args, { inspectorAdapters, abortSignal, getSearchSessionId, kibanaRequest }) {
+      if (!kibanaRequest) {
+        throw new Error(
+          i18n.translate('data.search.esaggs.error.kibanaRequest', {
+            defaultMessage:
+              'A KibanaRequest is required to execute this search on the server. ' +
+              'Please provide a request object to the expression execution params.',
+          })
+        );
+      }
```

Comment:
On the server we must have a `KibanaRequest` or we cannot scope the dependencies of `esaggs`, so any function that relies on a scoped service will need to throw if this handler is not present (since it's optional).